src/reducers/archiveThought.ts

```diff
@@ -123,24 +123,22 @@ const archiveThought = (state: State, { path }: { path: Path }): State => {
   }
 
   return reducerFlow([
-
-    // set the cursor away from the current cursor before archiving so that existingThoughtMove does not move it
-    setCursor({
-      path: cursorNew,
-      editing: state.editing,
-      offset,
-    }),
-
-    isDeletable
-      ? existingThoughtDelete({
-        context: showContexts ? context : parentOf(pathToContext(simplePath)),
-        showContexts,
-        thoughtRanked: head(simplePath),
-      })
-      : reducerFlow([
-
+    ...isDeletable
+      ? [
+        // set the cursor away from the current cursor before archiving so that existingThoughtMove does not move it
+        setCursor({
+          path: cursorNew,
+          editing: state.editing,
+          offset,
+        }),
```

Comment:
Why not call `setCursor` once at the tend, since it's the same regardless of `isDeletable`? It seems strange to duplicate this code in both cases, i.e.

```js
if (a) {
  b()
  c()
}
else {
  d()
  b()
}
```

Reply:
@raineorshine  I noticed that duplication earlier and tried moving the `setCursor` to the end. But some of the `archiveThoughts` tests start to break if we do that, so I believe that the order of execution is important here (as implied by [this comment](https://github.com/cybersemics/em/blob/3d3ed667401c08b2b95158037988eca51a9102c4/src/reducers/archiveThought.ts#L127)). 
```
// set the cursor away from the current cursor before archiving so that existingThoughtMove does not move it
```

Reply:
I did some more investigating and found the cause of the ordering issue. `existingThoughtMove` was not just setting the cursor incorrectly but crashing, so I suspected there was more to it. 

`archiveThought` was passing `SimplePaths` to `existingThoughtMove` instead of `Paths`, thus preventing the `cursor` updating logic from working. `existingThoughtMove` still needed `SimplePath` for some things (e.g. context, rank, and id), so I had to update it to do an internal `simplifyPath` conversion for just those particular cases, without changing the `Path` that was used for cursor calculations.